File: api/src/main/java/com/example/demo/model/DashboardDTO.java
package com.example.demo.model;

import lombok.Data;

import java.math.BigDecimal;
import java.util.List;

@Data
public class DashboardDTO {

    private BigDecimal custoPrevisto;
    private BigDecimal custoExecutado;
    private Double andamentoMediaGeral;
    private List<DashboardProjetoDTO> projetos;

}

File: ui/src/app/model/dashboard-dto.ts
import { FormControl, FormGroup, FormArray } from '@angular/forms';
import { DashboardProjetoDTO } from './dashboard-projeto-dto';

export class DashboardDTO {

  custoPrevisto: number | undefined;
  custoExecutado: number | undefined;
  andamentoMediaGeral: number | undefined;
  projetos: DashboardProjetoDTO [] | undefined;

  constructor() { 
    this.projetos = [];
  }

  static criarFormulario(dashboardDTO: DashboardDTO) {
    return new FormGroup({
      custoPrevisto: new FormControl(dashboardDTO.custoPrevisto),
      custoExecutado: new FormControl(dashboardDTO.custoExecutado),
      andamentoMediaGeral: new FormControl(dashboardDTO.andamentoMediaGeral),
      projetos: new FormArray(dashboardDTO!.projetos!.map(parcelaCompra => DashboardProjetoDTO.criarFormulario(parcelaCompra)))
    });
  }
}
